### PR TITLE
replacement of es5 functions to es6 classes featuring p5.delay

### PR DESCRIPTION
**Pull request by endurance21**
@therewasaguy  @kyle1james  please review this pR!

### CODE REVIEW

File: src/delay.js [hunk 1]
```diff
@@ -299,4 +298,15 @@ p5.Delay.prototype.dispose = function () {
...
+};
+
+
+
+
+
+
+
+
+
```
Comment (therewasaguy) [inline src/delay.js:311]: all these blank lines are unnecessary (another thing I wish the linter caught!)

File: src/delay.js [hunk 2]
```diff
@@ -124,7 +122,7 @@ p5.Delay.prototype = Object.create(Effect.prototype);
  *                               below the lowPass will be part of the
  *                               delay.
  */
-p5.Delay.prototype.process = function (src, _delayTime, _feedback, _filter) {
+process (src, _delayTime, _feedback, _filter) {
```
Comment (therewasaguy) [inline src/delay.js:125]: I think every line from here until the end of the `class` needs indentation to show that it's part of the `class`. Surprised our linter didn't catch this!

File: src/delay.js (hunk 1 above)
Comment (endurance21) [reply src/delay.js:311]: I think i should run the linter again
Will be reporting it soon

**Comment by endurance21** [comment]
@therewasaguy 
linter rocked this time ! 
i think its good to go !

**Pull request merged**